src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -388,12 +421,33 @@ def _create_request_body(
     """
     request_size_limit = max_error_size if any(map(is_span_has_error, msgs)) else max_size
 
+    size: int = get_event_base64_size(msgs)
     if not prune_size_flag or (
-        len(msgs) < NUMBER_OF_SPANS_IN_REPORT_OPTIMIZATION
-        and get_event_base64_size(msgs) < request_size_limit  # noqa
+        len(msgs) < NUMBER_OF_SPANS_IN_REPORT_OPTIMIZATION and size < request_size_limit  # noqa
     ):
         return aws_dump(msgs)[:request_size_limit]
 
+    """
+    Process spans: if should_try_zip is True, split and zip the spans, check their size,
+    and either return the zipped bulks or continue processing.
+    """
```

Comment:
We usually don't use strings for comments except for docstrings. Use `#` comment instead

Reply:
Done